File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p04000/p04000_num3_parsed.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int H, W, N;
    cin >> H >> W >> N;

    vector<vector<int>> grid(H, vector<int>(W, 0));

    for (int i = 0; i < N; i++) {
        int a, b;
        cin >> a >> b;
        grid[a - 1][b - 1] = 1;
    }

    vector<int> count(10, 0);

    for (int i = 0; i < H - 2; i++) {
        for (int j = 0; j < W - 2; j++) {
            int blacks = 0;
            for (int r = i; r < i + 3; r++) {
                for (int c = j; c < j + 3; c++) {
                    if (grid[r][c] == 1) {
                        blacks++;
                    }
                }
            }
            count[blacks]++;
        }
    }

    for (int i = 0; i <= 9; i++) {
        cout << count[i] << endl;
    }

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p04000/p04000_num3_parsed.py
import sys

def count_black_cells(H, W, N, cells):
    grid = [[0] * W for _ in range(H)]
    for a, b in cells:
        grid[a-1][b-1] = 1

    count = [0] * 10
    for i in range(H-2):
        for j in range(W-2):
            num_black = 0
            for x in range(i, i+3):
                for y in range(j, j+3):
                    if grid[x][y] == 1:
                        num_black += 1
            count[num_black] += 1
    
    return count

H, W, N = map(int, input().split())
cells = [list(map(int, input().split())) for _ in range(N)]

result = count_black_cells(H, W, N, cells)

for num in result:
    print(num)